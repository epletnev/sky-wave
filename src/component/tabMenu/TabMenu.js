import React, { useEffect } from "react";
import { useLocation } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { getMenu } from "../../redux/menu/menuAction";
import MenuCard from "../card/Card";
import "./TabMenu.scss";

function TabMenu({ category }) {
  const dispatch = useDispatch();
  const menu = useSelector((state) => state.menu.menu);
  const location = useLocation();

  useEffect(() => {
    dispatch(getMenu());
  }, []);

  return (
    <div className="flex-container">
      {location.pathname === "/"
        ? menu.map((item) => <MenuCard key={item.id} item={item} />)
        : menu
            .filter(
              (item) => category === "All" || item.categoryId === category
            )
            .map((item) => <MenuCard key={item.id} item={item} />)}
    </div>
  );
}

export default TabMenu;
