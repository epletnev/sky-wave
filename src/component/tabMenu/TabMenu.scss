@import "../../theme/variables.scss";

.flex-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
