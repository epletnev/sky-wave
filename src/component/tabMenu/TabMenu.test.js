import React from "react";
import TabMenu from "./TabMenu";

it("expect to render TabMenu component", () => {
  const mockMenu = [
    {
      id: 1,
      name: "Snack Set",
      description: "Butter, cream, honey, salty cottage cheese, white cheese.",
      price: 8.2,
      image: {
        small: "assets/images/foods/breakfast/1/small.jpg",
        medium: "assets/images/foods/breakfast/1/medium.jpg",
        big: "assets/images/foods/breakfast/1/big.jpg",
      },
      discount: 0,
      ratingsCount: 4,
      ratingsValue: 350,
      availibilityCount: 5,
      cartCount: 0,
      weight: 160,
      isVegetarian: false,
      categoryId: "Breakfast",
      reviews: [],
    },
  ];
  expect(<TabMenu menu={mockMenu} />).toMatchSnapshot();
});
