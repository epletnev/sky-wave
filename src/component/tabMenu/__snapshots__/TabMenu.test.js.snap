// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render TabMenu component 1`] = `
<TabMenu
  menu={
    Array [
      Object {
        "availibilityCount": 5,
        "cartCount": 0,
        "categoryId": "Breakfast",
        "description": "Butter, cream, honey, salty cottage cheese, white cheese.",
        "discount": 0,
        "id": 1,
        "image": Object {
          "big": "assets/images/foods/breakfast/1/big.jpg",
          "medium": "assets/images/foods/breakfast/1/medium.jpg",
          "small": "assets/images/foods/breakfast/1/small.jpg",
        },
        "isVegetarian": false,
        "name": "Snack Set",
        "price": 8.2,
        "ratingsCount": 4,
        "ratingsValue": 350,
        "reviews": Array [],
        "weight": 160,
      },
    ]
  }
/>
`;

exports[`expext to render TabMenu component 1`] = `
<TabMenu
  menu={
    Array [
      Object {
        "availibilityCount": 5,
        "cartCount": 0,
        "categoryId": "Breakfast",
        "description": "Butter, cream, honey, salty cottage cheese, white cheese.",
        "discount": 0,
        "id": 1,
        "image": Object {
          "big": "assets/images/foods/breakfast/1/big.jpg",
          "medium": "assets/images/foods/breakfast/1/medium.jpg",
          "small": "assets/images/foods/breakfast/1/small.jpg",
        },
        "isVegetarian": false,
        "name": "Snack Set",
        "price": 8.2,
        "ratingsCount": 4,
        "ratingsValue": 350,
        "reviews": Array [],
        "weight": 160,
      },
    ]
  }
/>
`;
